### DISCUSSION TITLE
Using a style selector which is targeted to every control

### DISCUSSION THREAD

**Discussion by HoneyTauOverTwo** [AvaloniaUI/Avalonia#15841] [Q&A] [answered]
I was looking for a way to set a margin value to every Control to a specific value. To accomplish that, I tried to creating a style on App.xaml: (the margin value is absurdly big so I could notice any change)

```
<Application.Styles>
	<FluentTheme />
	<Style Selector="Control">
		<Setter Property="Margin" Value="50"/>
	</Style>
</Application.Styles>
```
 
That did not work, I also tried using the selectors `*` and `Control.m` and applying the proper class to the control using `Classes="m"` which didn't work either. Using a specific control name like `TextBlock` works properly as expected.

Is there currently a way to accomplish this in Avalonia without having to create an exhaustive list of every control individually?

**Comment by ykarpeev** [comment]
What about this:

```
 <Application.Resources>
        <ControlTheme x:Key="theme1" TargetType="Control">
            <Setter Property="Margin" Value="50" />
        </ControlTheme>
    </Application.Resources>
```


and then  Theme="{StaticResource theme1}"> on the controls.

**Comment by stevemonaco** [comment, marked as answer]
Use the `:is` selector to match a type and all derived types: `<Style Selector=":is(Control)">`

Please see the docs: https://docs.avaloniaui.net/docs/reference/styles/style-selector-syntax#include-derived-classes

Whether this is a good or bad idea depends on exactly what you're trying to do because `Control` is extremely broad.

**Reply by HoneyTauOverTwo** [reply]
This indeed works. But as you expected, `Control` is too broad. Doing this:

```
<Style Selector=":is(Control)">
	<Setter Property="Margin" Value="50"/>
</Style>
```

Has proven that this applies margins on stuff that you wouldn't want.
